File: SpringBootServer/src/main/java/com/chamith/vote_streamer/model/ElectionResult.java
package com.chamith.vote_streamer.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
public class ElectionResult {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String electoralDistrict;
    private String pollingDivision;

    @OneToMany(mappedBy = "electionResult", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Party> parties;

    private int validVotes;
    private int polledVotes;
    private int registeredVotes;
    private LocalDateTime addedTime;

}

File: app/src/main/java/com/chamith/srilankavotestream/data/ElectionResult.kt
package com.chamith.srilankavotestream.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.time.LocalDateTime

@Parcelize
data class ElectionResult(

    val electoralDistrict: String,
    val pollingDivision: String,
    val addedTime: LocalDateTime

) : Parcelable
